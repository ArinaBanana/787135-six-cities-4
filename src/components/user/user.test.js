import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {MemoryRouter} from "react-router-dom";

import User from "./user.jsx";

const middlewares = [];
const mockStore = configureStore(middlewares);

describe(`Should render User`, () => {
  it(`Should render User with false isAuth`, () => {
    const store = mockStore({
      USER: {
        user: null,
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <MemoryRouter>
            <User isAuth={false} />
          </MemoryRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should render User with true isAuth`, () => {
    const store = mockStore({
      USER: {
        user: {
          email: `email`
        },
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <MemoryRouter>
            <User isAuth={true} />
          </MemoryRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
