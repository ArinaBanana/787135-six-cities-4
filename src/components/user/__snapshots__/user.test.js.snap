// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render User Should render User with false isAuth 1`] = `
<li
  className="header__nav-item user"
>
  <a
    className="header__nav-link header__nav-link--profile"
    href="/login"
    onClick={[Function]}
  >
    <div
      className="header__avatar-wrapper user__avatar-wrapper"
    />
    <span
      className="header__login"
    >
      Sign in
    </span>
  </a>
</li>
`;

exports[`Should render User Should render User with true isAuth 1`] = `
<li
  className="header__nav-item user"
>
  <a
    className="header__nav-link header__nav-link--profile"
    href="/login"
    onClick={[Function]}
  >
    <div
      className="header__avatar-wrapper user__avatar-wrapper"
    />
    <span
      className="header__login"
    >
      Sign in
    </span>
  </a>
</li>
`;
